### PR TITLE
Add EDNS client tag functionality

### PR DESCRIPTION
**Pull request by ralphdolmans**
Add option to include 16 bit EDNS tags in upstream queries, for specified destination addresses. Max one tag per query, the most specific match is selected.

Tags are configured using the `edns-client-tag: <IP netblock> <tag data>` configuration option, which is allowed multiple times. Tag data is configured in decimal format.

### CODE REVIEW

File: util/edns.h [hunk 1]
```diff
@@ -42,11 +42,66 @@
...
+	uint16_t tag_data;
+};
+
+/**
+ * Create structure to hold EDNS tags
+ * @return: newly created edns_tags, NULL on alloc failure.
+ */
+struct edns_tags* edns_tags_create(void);
+
+/** Delete ENDS tags structure
```
Comment (wcawijngaards) [inline util/edns.h]: s/ENDS/EDNS/

File: util/module.h [hunk 2]
```diff
@@ -520,6 +520,7 @@ struct module_env {
 	struct edns_known_option* edns_known_options;
 	/* Number of known edns options */
 	size_t edns_known_options_num;
+	struct edns_tags* edns_tags;
```
Comment (wcawijngaards) [inline util/module.h:524]: A comment line above the member, like the other entries.  `/** EDNS client tag information */` or so.

File: util/edns.h (hunk 1 above)
Comment (ralphdolmans) [reply util/edns.h]: thanks, fixed

File: util/module.h (hunk 2 above)
Comment (ralphdolmans) [reply util/module.h:524]: thanks, fixed

**Review by wcawijngaards — APPROVED** [review]
Review looks fine for me.

**Pull request merged**